packages/explorer-helpers/src/constructMultipartForm.ts

```diff
@@ -0,0 +1,203 @@
+import type { JSONObject } from './types';
+
+export type FileVariable = {
+  variableKey: string;
+  files: File[] | undefined;
+  isMultiFile: boolean;
+};
+
+export interface OperationRequestBody {
+  query: string;
+  variables: JSONObject | null;
+  operationName: string | undefined;
+}
+
+// https://github.com/apollographql/apollo-client/blob/cbcf951256b22553bdb065dfa0d32c0a4ca804d3/src/link/http/serializeFetchParameter.ts
+export const serializeFetchParameter = (p: any, label: string) => {
+  let serialized;
+  try {
+    serialized = JSON.stringify(p);
+  } catch (e) {
+    const parseError = new Error(
+      `Network request failed. ${label} is not serializable`
+    );
+    throw parseError;
+  }
+  return serialized;
+};
+
+const isFileVariableWithFile = (
+  fileVariable: FileVariable
+): fileVariable is FileVariable & {
+  files: NonNullable<FileVariable['files']>;
+} => {
+  return !!fileVariable.files && fileVariable.files.length > 0;
+};
+
+export type FileVariableFromTransfer = {
+  variableKey: string;
+  files: { arrayBuffer: ArrayBuffer; fileName: string }[];
+  isMultiFile: boolean;
+};
+
+export const getFileListFromTransfer = ({
+  fileVariables: inputtedFileVariables,
+}: {
+  fileVariables: FileVariableFromTransfer[];
+}) =>
+  inputtedFileVariables.map((fileVariable) => ({
+    ...fileVariable,
+    files: fileVariable.files.map(
+      ({ arrayBuffer, fileName }) =>
+        new File([new Blob([arrayBuffer])], fileName)
+    ),
+  }));
+
+const getSharedMultipartFormComponents = ({
```

Comment:
we'll need to export this yeah? 
```suggestion
export const getSharedMultipartFormComponents = ({
```

Reply:
Oooh no I see, you moved curl in here as well 👍

Reply:
We only use it in this file, we don't use it in explorer or the embeds otherwise! And we export the 2 functions that use it 👍🏻